react-native/react/constants/platform.native.desktop.js

```diff
@@ -1,7 +1,39 @@
 'use strict'
 
 import {OS_DESKTOP} from './platform.shared'
+import path from 'path'
 
 export const isDev = process.env.NODE_ENV === 'development'
 export const OS = OS_DESKTOP
 export const isMobile = false
+
+const runMode = process.env.KEYBASE_RUN_MODE || 'devel'
+const envedPathOSX = {
+  staging: 'KeybaseStaging',
+  devel: 'KeybaseDevel',
+  prod: 'Keybase'
+}
+
+function findSocketRoot () {
+  const paths = {
+    'darwin': `${process.env.HOME}/Library/Caches/${envedPathOSX[runMode]}/`,
+    'linux': `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`
```

Comment:
@chrisnojima came over to me.  Plan:

* core will move socket to XDG_DATA_DIR
* then we'll have another patch against this code to use XDG_DATA_DIR instead of XDG_RUNTIME_DIR

:thumbsup: